packages/lisk-transactions/test/transactions/base.ts

```diff
@@ -0,0 +1,746 @@
+			const invalidTransaction = {
+				...defaultTransaction,
+				amount: 0,
+				fee: 'fee',
+			};
+
+			it('should throw a transaction multierror', () => {
+				try {
+					new TestTransaction(
+						(invalidTransaction as unknown) as TransactionJSON,
+					);
+				} catch (error) {
+					return expect(error).to.be.an.instanceOf(TransactionMultiError);
+				}
+
+				return Promise.resolve();
+			});
+		});
+	});
+
+	describe('#assetToJSON', () => {
+		it('should return an object of type transaction asset', () => {
+			return expect(baseTransaction.assetToJSON()).to.be.an('object');
+		});
+	});
+
+	describe('#toJSON', () => {
+		it('should return transaction json', () => {
+			baseTransaction = new TestTransaction(defaultTransaction);
+			const transactionJSON = baseTransaction.toJSON();
+
+			return expect(transactionJSON).to.be.eql(defaultTransaction);
+		});
+	});
+
+	describe('#getBytes', () => {
+		it('should return a buffer', () => {
+			expect(baseTransaction.getBytes()).to.be.an.instanceOf(Buffer);
+		});
+	});
+
+	describe('#containsUniqueData', () => {
+		it('should return a boolean', () => {
+			expect(baseTransaction.containsUniqueData()).to.be.a('boolean');
+		});
+	});
+
+	describe('#checkSchema', () => {
+		describe('when given valid transaction', () => {
+			beforeEach(() => {
+				baseTransaction = new TestTransaction(defaultTransaction);
+
+				return Promise.resolve();
+			});
+
+			it('should return a successful transaction response', () => {
+				const { id, status, errors } = baseTransaction.checkSchema();
+
+				expect(id).to.be.eql(baseTransaction.id);
+				expect(errors).to.be.eql([]);
+				return expect(status).to.eql(Status.OK);
+			});
+		});
+
+		describe('when given invalid data', () => {
+			let invalidTransaction: any;
+			beforeEach(() => {
+				invalidTransaction = {
+					type: 0,
+					amount: '00001',
+					fee: '0000',
+					recipientId: '',
+					senderPublicKey: '11111111',
+					senderId: '11111111',
+					timestamp: 79289378,
+					asset: {},
+					signature: '1111111111',
+					id: '1',
+				};
+				baseTransaction = new TestTransaction(invalidTransaction as any);
+				return Promise.resolve();
+			});
+
+			describe('when checked against baseTransaction schema validator', () => {
+				it('should return a failed transaction response', () => {
+					const { id, status, errors } = baseTransaction.checkSchema();
+					const errorsArray = errors as ReadonlyArray<TransactionError>;
+
+					expect(id).to.be.eql(baseTransaction.id);
+					errorsArray.forEach(error =>
+						expect(error).to.be.instanceof(TransactionError),
+					);
+					return expect(status).to.eql(Status.FAIL);
+				});
+			});
+
+			describe('when given unmatching senderId and senderPublicKey', () => {
+				beforeEach(() => {
+					const invalidIdTransaction = {
+						...defaultTransaction,
+						senderId: defaultTransaction.senderId.replace('1', '0'),
+					};
+
+					baseTransaction = new TestTransaction(invalidIdTransaction as any);
```

Comment:
Same goes here as well. I think for this test, the `new Transaction` will be called 5 times.